avro-java8/src/main/java/com/fasterxml/jackson/dataformat/avro/java8/deser/BaseTimeJsonDeserializer.java

```diff
@@ -0,0 +1,43 @@
+package com.fasterxml.jackson.dataformat.avro.java8.deser;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.TimeUnit;
+
+public abstract class BaseTimeJsonDeserializer<T> extends JsonDeserializer<T> {
```

Comment:
I'd suggest extending `StdScalarDeserializer` instead.

Reply:
Also, on naming, since this is Avro-specific, maybe rather BaseTimeAvroDeserializer (or something).